spec/support/comment_shared_contexts.rb

```diff
@@ -0,0 +1,12 @@
+RSpec.shared_context "comment resource helpers" do
+  let(:resource_name) { subject.comment_on_type.downcase.pluralize }
+  let(:resource_id) { subject.comment_on_id }
+
+  def new_comment_path(options = subject.zhishi_url_options)
+    comments_path(options[:resource_name], options[:resource_id])
```

Comment:
I think this would still do the same thing, rather than splitting them up again:
```ruby
    comments_path(subject.zhishi_url_options)
```
Since, it's a hash